components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -239,16 +241,21 @@ class GeckoWebExtension(
     /**
      * See [WebExtension.registerTabHandler].
      */
-    override fun registerTabHandler(tabHandler: TabHandler) {
-
+    override fun registerTabHandler(tabHandler: TabHandler, defaultSettings: Settings?) {
         val tabDelegate = object : GeckoNativeWebExtension.TabDelegate {
 
             override fun onNewTab(
                 ext: GeckoNativeWebExtension,
                 tabDetails: GeckoNativeWebExtension.CreateTabDetails
             ): GeckoResult<GeckoSession>? {
-                val geckoEngineSession = GeckoEngineSession(runtime, openGeckoSession = false)
+                val trackingPolicy = defaultSettings?.trackingProtectionPolicy
+                    ?: EngineSession.TrackingProtectionPolicy.strict()
 
+                val geckoEngineSession = GeckoEngineSession(
+                    runtime,
+                    defaultSettings = DefaultSettings(trackingProtectionPolicy = trackingPolicy),
```

Comment:
Ideally we just pass the same instance of settings along here. See my comment above.

Reply:
@codrut-topliceanu we are still creating a new settings object, let just pass the the reference (`defaultSettings`) that we get `registerTabHandler`.

Reply:
Creating a new object, we are discarding all the previous and only keeping the `trackingProtectionPolicy`

Reply:
I think I'm blind. Thank you!